### PR TITLE
User profile changes

### PR DESCRIPTION
**Pull request by mctian**

### CODE REVIEW

**Comment by mctian** [comment]
Also fixes undefined link display on profile page.

**Comment by gezhao** [comment]
I saw you add solution recommendation there. Can you tell the changes in plain English here?

**Comment by mctian** [comment]
Main change is white papers are now also recommended. I still need to filter out non-solution related materials from ppr, but the majority of recommended materials are either articles or white papers.
